components/Footer.tsx

```diff
@@ -0,0 +1,34 @@
+type PropsType = {
+    viewCart:boolean
+}
+
+const footer = ({viewCart}:PropsType) => {
+  
+    const year : number = new Date().getFullYear()
+    
+    const content = (
+        <footer className="min-h-screen flex flex-col justify-between ">
+            {viewCart ? (
+                <div className="absolute inset-x-0 bottom-0  p-6 flex items-center justify-center text-lg "> 
+                Shopping-cart &copy; {year}</div>
```

Comment:
Shopping-cart 2024 está a ser repetido tanto aqui como na linha 22. 
Podia-se colocá-lo abaixo do if then else para não ser repetido e trocar-se o if then else por um 'not' viewCart && (o que está no else).
Acho que assim ficava mais simples.